static/src/javascripts/projects/commercial/modules/dfp/on-slot-render.js

```diff
@@ -74,7 +74,14 @@ export const onSlotRender = (event: SlotRenderEndedEvent): void => {
         reportEmptyResponse(advert.id, event);
         emitRenderEvents(false);
     } else {
-        advert.size = event.size;
+        /**
+         * if advert.hasPrebidSize is false we use size
+         * from the GAM event when adjusting the slot size.
+         * */
+        if (!advert.hasPrebidSize) {
+            advert.size = event.size;
```

Comment:
A test of this might help mitigate future-confusion, particularly if it could test the full refresh lifecycle.

Reply:
Testing the full refresh lifecycle is a bit of a black box once it enter the `window.googletag.pubads` event emitters, I can write a test based on what we've seen - a `window.pbjs.renderAd()` call trigger `prebid`s `bidWon` event which precedes `onSlotRender`.